flink-runtime/src/main/java/org/apache/flink/runtime/state/filemerging/EmptyFileMergingOperatorStreamStateHandle.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.runtime.state.filemerging;
+
+import org.apache.flink.runtime.state.OperatorStateHandle;
+import org.apache.flink.runtime.state.StreamStateHandle;
+
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * An empty {@link FileMergingOperatorStreamStateHandle} that is only used as a placeholder to
+ * prevent file merging directory from being deleted.
+ */
+public class EmptyFileMergingOperatorStreamStateHandle
```

Comment:
I'm not sure if `EmptyFileMergingOperatorStreamStateHandle` should extend the `FileMergingOperatorStreamStateHandle` since it does not need the delegating functionality of `OperatorStreamStateHandle`. Maybe directly built on top of `OperatorStateHandle` is enough?

Reply:
`OperatorStateHandle`  requires the implementation of `getStateNameToPartitionOffsets()` and `getDelegateStateHandle()`. If `EmptyFileMergingOperatorStreamStateHandle ` directly inherits `OperatorStateHandle `, we need to repeatedly implement the methods already in `FileMergingOperatorStreamStateHandle`. I lean towards maintaining the current implementation.